packages/next/build/index.ts

```diff
@@ -442,11 +447,13 @@ export default async function build(dir: string, conf = null): Promise<void> {
         : path.join('static', buildId, 'pages', file)
       ).replace(/\\/g, '/')
 
-      let page = file.split('.html')[0].replace(/\\/g, '/')
-      pagesManifest[page] = relativeDest
-      page = page === '/index' ? '/' : page
-      pagesManifest[page] = relativeDest
-      staticPages.add(page)
+      if (!file.endsWith('.prerender.html')) {
```

Comment:
Seems weird that we rely on the extension of the file here instead of what we already know as truth, the `staticPages: Set`.

Maybe all of the logic in this branch can be reworked?

Reply:
Updated